File: src/main/java/com/github/zxh/classpy/classfile/bytecode/InvokeDynamic.java
package com.github.zxh.classpy.classfile.bytecode;

import com.github.zxh.classpy.classfile.constant.ConstantPool;
import com.github.zxh.classpy.classfile.jvm.Opcode;

public class InvokeDynamic extends Instruction {

    {
        u1  ("opcode");
        u2cp("index");
        u2  ("zero");
    }

    public InvokeDynamic(Opcode opcode, int pc) {
        super(opcode, pc);
    }
    
    @Override
    protected void postRead(ConstantPool cp) {
        setDesc(getDesc() + " " + super.get("index").getDesc());
    }
    
}

File: src/main/kotlin/com/github/zxh/classpy/classfile/bytecode/InvokeDynamic.kt
package com.github.zxh.classpy.classfile.bytecode

import com.github.zxh.classpy.classfile.constant.ConstantPool

class InvokeDynamic(opcode: Opcode, pc: Int) : Instruction(opcode, pc) {

    init {
        u1  ("opcode");
        u2cp("index");
        u2  ("zero");
    }

    override fun afterRead(cp: ConstantPool) {
        val operand = super.get("index")!!
        desc = "${opcode.name} ${operand.desc}"
    }
    
}
